x/evidence/keeper/infraction_test.go

```diff
@@ -1,15 +1,68 @@
 package keeper_test
 
 import (
-	"time"
-
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/runtime"
+	simtestutil "github.com/cosmos/cosmos-sdk/testutil/sims"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	authkeeper "github.com/cosmos/cosmos-sdk/x/auth/keeper"
+	bankkeeper "github.com/cosmos/cosmos-sdk/x/bank/keeper"
+	"github.com/cosmos/cosmos-sdk/x/evidence/keeper"
+	"github.com/cosmos/cosmos-sdk/x/evidence/testutil"
 	"github.com/cosmos/cosmos-sdk/x/evidence/types"
+	minttypes "github.com/cosmos/cosmos-sdk/x/mint/types"
+	slashingkeeper "github.com/cosmos/cosmos-sdk/x/slashing/keeper"
 	"github.com/cosmos/cosmos-sdk/x/staking"
+	stakingkeeper "github.com/cosmos/cosmos-sdk/x/staking/keeper"
 	"github.com/cosmos/cosmos-sdk/x/staking/teststaking"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	"time"
 )
 
-func (suite *KeeperTestSuite) TestHandleDoubleSign() {
+type InfractionTestSuite struct {
+	suite.Suite
```

Comment:
I am not exactly sure, why use the mixing of mock test and integration test?
For example, why are you using simtestuitl instead of KeeperTestSuite?

Reply:
Because this seems to be more like an integration test, and logic is a bit critical to just mock, it felt more like I was testing mock expect than anything else. We discuss it a bit and for now we left it like that

Reply:
In this case, we don't use https://github.com/cosmos/cosmos-sdk/blob/035402d2ea73777e5d986b9f39fa8b0369a8e790/x/evidence/keeper/keeper_test.go#L145 anywhere

Reply:
true sorry I missed that thanks!!!